docs/source/configuration/validation.md

```diff
@@ -0,0 +1,273 @@
+
+```ts
+let blockSchema = {
+  // ... fieldset definition in here
+  properties: {
+    ...schema.properties,
+    customField: {
+      title: 'My custom URL field',
+      description: '',
+      format: 'url'
+    },
+  },
+  required: [],
+};
+```
+
+`url` named validator should be registered by this name as a Volto validator utility:
+
+```ts
+config.registerUtility({
+  type: 'validator',
+  name: 'url',
+  method: urlValidator,
+})
+```
+
+In this case, the `urlValidator` method validator will be applied for the block field `customField`.
+
+#### For content types
+
+Content types also can specify the `format` using the schema hints in the backend in the `frontendOptions`:
+
+```python
+from plone.supermodel import model
+from zope import schema
+
+class IMyContent(model.Schema):
+    directives.widget(
+        "customField",
+        frontendOptions={
+            "format": "url",
+        },
+    )
+    customField = schema.TextLine(
+        title="Custom URL field",
+        required=False,
+    )
+    # Rest of your content type definition
+```
+
+For the record, the resultant `plone.restapi` response will be something like the following, a bit different than in blocks JSON schema.
+But the validation engine will take care too:
+
+```json
+{
+  "properties": {
+    "customField": {
+      "title": "Custom URL field",
+      "widgetOptions": {
+        "frontendOptions": {
+          "format": "url"
+        }
+      }
+    }
+  }
+}
+```
+
+and the `urlValidator` method validator will be applied for the content type field `customField`.
+
+### Advanced scenarios
+
+In case you need more granularity and you don't have access to modify the existing implementation of the JSON schema definitions for existing content types, blocks or forms (might be in third party add-ons), you can use the following advanced validator registrations, using `field`, `widget`, `behaviorName` or `blockType` validator registrations.
+
+#### Per field `type` validators
+
+These validators are applied depending on the specified `type` of the field in the JSON schema from content types, forms or blocks.
+The next example is for the use case of JSON schema defined in a block:
+
+```ts
+let blockSchema = {
+  // ... fieldset definition in here
+  properties: {
+    ...schema.properties,
+    customField: {
+      title: 'My custom field',
+      description: '',
+      type: 'integer',
+      maximum: 30
+    },
+  },
+  required: [],
+};
+```
+
+```ts
+config.registerUtility({
+  type: 'validator',
+  name: 'maximum',
+  dependencies: {
+    fieldType: 'integer',
+  },
+  method: maximumValidator,
+})
+```
+
+You should specify the `type` in the JSON schema of the block (in a content type, it is included in the default serialization of the field).
+If a field does not specify type, it assumes a `string` type as validator.
+
+#### Per field `widget` validators
+
+These validators are applied depending on the specified `widget` of the field.
```

Comment:
In the above examples you have to specify in the json schema if the validator is enabled or not. In the registerUtility method you only specify if the validator is available or not. It seems in this example registering the validator will have it enabled by default. I think for consistency we should also specify it in the schema and not enable it by default.

Reply:
I will clarify which validators are included and enabled by default, and put it on the top, instead than at the bottom.

Reply:
Yeah, having that at the end of the docs, with a `seealso` where it was first mentioned, didn't feel right. This would be a good change.

Reply:
@stevepiercy moved to the top.